lib/etcdctl.py

```diff
@@ -0,0 +1,106 @@
+from charmhelpers.core.hookenv import log
+from subprocess import CalledProcessError
+from shlex import split
+from subprocess import check_output
+import os
+
+
+class EtcdCtl:
+    ''' etcdctl modeled as a python class. This python wrapper consumes
+    and exposes some of the commands contained in etcdctl. Related to unit
+    registration, cluster health, and other operations '''
+
+    def register(self, cluster_data):
+        ''' Perform self registration against the etcd leader and returns the
+        raw output response.
```

Comment:
The method does not return raw output, the docstring is not accurate. Suggest  ... and returns None if registration fails, or a registration dictionary with the parsed cluster_unit_id, and initial cluster string.